keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1073 @@
+- Check the authorization mode for existing pods. For example, if a pod’s service account is not
+  authorized to use the PSP that validated it (based on the `kubernetes.io/psp` annotation), then
+  that should trigger a warning.
+- Automate the dry-run and/or labeling process.
+- Automatically select (and optionally apply) a policy level for each namespace.
+
+We should also publish a step-by-step migration guide. A rough approach might look something like
+this, with the items tagged (automated) having support from the PSP migration tool.
+
+1. Enable 3-tier policy admission plugin, default everything to privileged.
+2. Eliminate mutating PSPs:
+    1. Clone all mutating PSPs to a non-mutating version (automated)
+    2. Update all ClusterRoles authorizing use of the mutating PSPs to also authorize use of the
+       non-mutating variant (automated)
+    3. Watch for pods using the mutating PSPs (check via the `kubernetes.io/psp` annotation), and
+       work with code owners to migrate to valid non-mutating resources.
+    4. Delete mutating PSPs
+3. Select a compatible profile for each namespace, based on the existing resources in the namespace
+   (automated)
+    1. Review the profile choices
+    2. Evaluate the difference in privileges that would come from disabling the PSP controller
+       (automated).
+4. (optional) Apply the profiles in `warn` and `audit` mode (automated)
+5. Apply the profiles in `enforce` mode (automated)
+6. Disable PodSecurityPolicy
+
+<<[/UNRESOLVED]>>
+
+### Graduation Criteria
+
+Maturity level of this feature is defined by:
+- `PodSecurity` feature gate
+- Documented maturity of the feature repo (library & webhook implementations)
+
+#### Alpha
+
+The initial alpha implementation targetting v1.22 includes:
+
+- Initial implementation is protected by the default-disabled feature gate `PodSecurity`
+- The built-in PodSecurity admission controller is defalut-disabled.
+- Initial set of E2E feature tests implemented and enabled in an alpha test job
+
+#### Beta
+
+We are targeting Beta in v1.23.
+
+1. Resolve the following sections:
+    - [ ] [Restricted policy support for Windows pods](#windows-restricted-profile-support)
+    - [ ] [Deprecation / removal policy for old profile versions](#versioning)
+    - [ ] [Ephemeral containers support](#ephemeral-containers)
+    - [ ] [PSP migration workflow & support](#podsecuritypolicy-migration)
+2. Collect feedback from the alpha, analyze usage of the webhook implementation.
+3. Thorough testing is already expected for alpha, but we will review our test coverage and fill any
+   gaps prior to beta.
+   - Feature tests are moved to the main test jobs (may postpone to GA)
+4. Admission plugin included in the default enabled set (enforcement is still opt-in per-namespace).
+
+#### GA
+
+<<[UNRESOLVED]>>
+
+We are targeting GA in v1.24 to allow for migration off PodSecurityPolicy before it is removed in
+v1.25.
```

Comment:
Yeah. We shouldn't rush it just for the sake of rushing it. If there are red flags, we'll hold it back. PSP is only beta, so I don't think it would be terrible if this was still beta in v1.24 (although I'd really like to get it past alpha).